src/mfs.jl

```diff
@@ -0,0 +1,163 @@
+
+""" 
+    check_basin(mapper, X, Y)
+
+Input:
+    mapper: AttractorMapper
+    X: Point 1
+    Y: Point 2
+Output: 
+    true if X and Y are in the same basin of attraction, false otherwise
+
+Description:
+    This function checks if two points are in the same basin of attraction. 
+    It does so by checking if the points are mapped to the same point by the mapper.
+
+"""
+
+function check_basin(mapper::AttractorMapper, X, Y)
+    idx = mapper(X)
+    idy = mapper(Y)
+    if idx == idy
+        return true
+    else
+        return false
+    end
+end
+
+
+""" 
+    random_point_on_sphere(radius, n)
+
+Input:
+    radius: radius of the sphere
+    n: dimension of the sphere
+Output:
+    coordinates of a random point on the n-sphere
+
+Description:
+    This function generates a random point on the n-sphere of radius radius.
+    It does so by generating n-1 random angles and then calculating the last coordinate
+    using the formula for the n-sphere.
+"""
+
+
+function random_point_on_sphere(radius, n)
```

Comment:
There is a more efficient way to do this, that we use in StateSpaceSets.jl, in the sampler function: https://github.com/JuliaDynamics/StateSpaceSets.jl/blob/0f0e5224e3d7b2922bb6e408911ad932cbe22ef5/src/sampler.jl#L88-L103 You should utilize the `statespace_sampler` function directly so that we reduce the amount of total code written. You can re-initialize a random sampler via `statespace_sampler` each time the radius changes.